com.unity.render-pipelines.high-definition/Editor/RenderPipeline/HDRenderPipelineUI.cs

```diff
@@ -411,15 +411,26 @@ static void Drawer_SectionDecalSettings(SerializedHDRenderPipelineAsset serializ
                 if (EditorGUI.EndChangeCheck())
                     serialized.renderPipelineSettings.decalSettings.atlasHeight.intValue = Mathf.Max(serialized.renderPipelineSettings.decalSettings.atlasHeight.intValue, 0);
 
+                EditorGUI.BeginChangeCheck();
                 EditorGUILayout.PropertyField(serialized.renderPipelineSettings.decalSettings.perChannelMask, Styles.metalAndAOContent);
+                if (EditorGUI.EndChangeCheck())
+                {
+                    // Tell VFX
+                    ((HDRenderPipelineEditor) owner).needRefreshVfxWarnings = true;
+                }
```

Comment:
Use the scoped version:
```
using (var scope = new EditorGUI.ChangeCheckScope())
                {
                    EditorGUILayout.PropertyField(serialized.renderPipelineSettings.decalSettings.perChannelMask, Styles.metalAndAOContent);
                    // Tell VFX
                    if (scope.changed)
                        ((HDRenderPipelineEditor)owner).needRefreshVfxWarnings = true;
                }
```

Reply:
Hi Alex, thanks for your feedback, greatly appreciated !
Concerning this one, I believe that the rest of the files uses the Begin/End pattern for performance/allocs reasons.

Reply:
I think is more readable and the code is not at Runtime. But this is up to you :)